furnace.c

```diff
@@ -56,6 +56,28 @@ typedef struct {
   #define MAPPER_MINUTES 1
 #endif
 
+#if CONFIG_SHUTTER
+typedef struct {
+  absolute_time_t deadline;
+  uint16_t        time_ms;
+  uint8_t         intern_state;
+} shutter_context_t;
+
+#define MAX_SHUTTER_MS 60000
+
+#define SHUTTER_DELAY_MS  250             // delay for shutter to get to right position
+
+#define SHUTTER_START_UNSTABLE    0
+#define SHUTTER_END_UNSTABLE      1
+#define SHUTTER_START_STABLE      2
+#define SHUTTER_END_STABLE        3
+#define SHUTTER_ON_OPTION         4
+#define SHUTTER_OFF_OPTION        5
+
+#define SHUTTER_PIN 0
+#define SHUTTER_PIN_SLICE pwm_gpio_to_slice_num(SHUTTER_PIN)
+#endif
+
```

Comment:
Can we move this to a separate file `shutter.c` and include it here? Same goes with `do_shutter_work()`.